scripts/generate_tokens.mjs

```diff
@@ -0,0 +1,137 @@
+// Purpose:
+//   to generate the files for each token is assetlists.
+
+// Requirements:
+//   we need to name the file: { toLower(asset.symbol) }_token_info_en.json
+//   we need to pull: asset.name, asset.symbol, asset.description, asset.coingecko_id
+//   but if the asset is a Chain's staking asset,
+//     we also pull: chain.website, chain.description
+//       then we combine asset.description and chain.description into a single description
+//     twitterURL will be scraped from coingecko, but if it's already here, we shall not remove it.
+
+// example output file:
+// {
+//   "localization": "en",
+//   "name": "Akash",
+//   "symbol": "AKT",
+//   "description": "Akash is open-source Supercloud that lets users buy and sell computing resources securely and efficiently. Purpose-built for public utility.",
+//   "websiteURL": "https://akash.network/",
+//   "twitterURL": "https://twitter.com/akashnet_",
+//   "coingeckoID": "akash-network"
+// }
+
+// -- THE PLAN --
+//
+//  iterate through the list of assets in assetlist (read directory should be a global var)
+//  for each asset,
+//    create an object, with properties for the file name and for the expected properties,
+//    compose the output file name ({ toLower(asset.symbol) }_token_info_en.json)
+//    see if the output file already exists, if so, pull the current property values into the object
+//    for each property,
+//      try to pull the value from the assetlist asset object and overwrite the current values
+//  serialize the object into JSON
+//  write JSON to file (file name is known, output directory should be a global var)
+//
+
+
+import * as fs from 'fs';
+import * as path from 'path';
+
+
+const tokenFileDir = "../contents/tokens/";
+const assetlistDir = "../../assetlists/osmosis-1/";

```

Comment:
Is it a file present in another repo?

Reply:
Yes, at the moment, this script would only run on my local PC where I have two repos 'token-info' and 'assetlists' in the same directory. Perhaps it could be adapted to run here if the repo were added as a submodule or something, but for now this script is only here for organization--making sure I don't lose the script--and in case anyone wants to see the logic of how it's generated.

Reply:
Sounds good